csrc/ir/utils.cpp

```diff
@@ -1400,22 +1417,37 @@ bool matchMatmulBiasCast(
   mma_lhs = mma->inA();
   mma_rhs = mma->inB();
 
-  // The expected ops for bias are: CastOp(bias, fp32) -> Broadcast -> Add
-  bias = binary->input(1); // Broadcasted bias tensor in fp32
-  auto* bcast = dynamic_cast<BroadcastOp*>(bias->definition());
-  NVF_ERROR(bcast != nullptr, "Expected bias tensor to be broadcasted.");
+  // Based on the presence of beta parameter, the expected ops are:
+  // CastOp(bias, fp32) -> -> Broadcast (Optional) -> Mul (if beta is present)
+  // -> Add
 
-  NVF_ERROR(
-      bias->as<TensorView>()->getRootDomain().back()->isBroadcast(),
-      "Expected last dimension to be broadcasted for bias tensor.");
+  // Check for beta parameter
+  auto* mul_beta = dynamic_cast<BinaryOp*>(binary->input(1)->definition());
+  if (mul_beta != nullptr && mul_beta->getBinaryOpType() == BinaryOpType::Mul) {
+    // Case 1: beta * C
+    beta = mul_beta->input(0);
+    bias = mul_beta->input(1);
+  } else {
+    // Case 2: C
+    bias = binary->input(1); // Broadcasted bias tensor in fp32
+  }
+
+  // Check if bias was broadcasted
+  auto* bcast = dynamic_cast<BroadcastOp*>(bias->definition());
+  if (bcast != nullptr) { // mma_lhs is broadcasted
+    // Bias of shape [M, 1]
+    NVF_ERROR(
+        bias->as<TensorView>()->getRootDomain().back()->isBroadcast(),
+        "Expected last dimension to be broadcasted for bias tensor.");
+    bias = bcast->input(0); // Bias tensor in fp32
+  }
 
-  bias = bcast->input(0); // Bias tensor in fp32
   auto* bias_cast = dynamic_cast<UnaryOp*>(bias->definition());
 
   // The bias tensor and matmul inputs should be of the same dtype.
   NVF_ERROR(
       bias_cast == nullptr || bias_cast->getUnaryOpType() == UnaryOpType::Cast,
-      "Expected the bias tensor to be preceded by castOp before broadcasting.");
+      "Expected the bias tensor to casted to Float.");
```

Comment:
I might have already asked this, but does the `bias` tensor have to be same type as `A` and `B` for `at::addmm`?

Reply:
Yes, the dtypes need to be the same